src/main.rs

```diff
@@ -42,48 +51,50 @@ fn main() {
     let path = Path::new(m.value_of("path").unwrap());
     let verbose = m.is_present("verbose");
 
-    let mut ret_val: i32 = 0;
+    let mut ret_val = ErrorCode::Ok;
 
     let batch: Vec<PathBuf> = if path.is_file() {
         vec![path.to_path_buf()]
     } else if path.is_dir() {
         match walk_indentex_files(&path) {
             Ok(b) => b,
             Err(e) => {
-                ret_val = 2;
+                ret_val = ErrorCode::WalkError;
                 println!("{}", Red.bold().paint(format!("{}", e)));
                 Vec::new()
             }
         }
     } else {
-        ret_val = 4;
+        ret_val = ErrorCode::FileTypeError;
         println!("{}",
                  Red.bold().paint(format!("Error: path '{}' is neither a file nor a directory",
                                           path.display())));
         Vec::new()
     };
 
-    for p in &batch {
-        if verbose {
-            print!("Transpiling file '{}'... ", p.display());
-        }
-        match transpile_file(&p) {
+    let ret_val_transpilation = batch.par_iter()
+        .map(|p| match transpile_file(&p) {
             Ok(_) => {
                 if verbose {
-                    println!("{}", Green.paint("ok"));
+                    println!("Transpiling file '{}'... {}",
+                             p.display(),
+                             Green.paint("ok"));
                 }
+                return ErrorCode::Ok as i32;
```

Comment:
I think it is more idiomatic to do like this: `return ErrorCode::Ok as i32;` :arrow_right: `ErrorCode::Ok` (no semicolon) here...